[Pages](docs): Update Remix deployment guide to reflect Remix Vite setup

AdiRishi opened the pull request:
Resolves #13820

Comments on the changes I've made
- I've paid special attention to ensuring each part of the page is accurate, I've run all the commands mentioned and double checked they all work with the latest version of Remix
- I updated the final example on the page from KV + DO to D1 instead. I made this decision because anecdotally speaking the most questions I've seen in the CF Discord server have always been around D1, it seems to be the more popular resource to access with Pages. 
- When describing how to fix types for bindings I am very explicit and show the entire code as an example. However, I could be more brief and link to the [Remix docs](https://remix.run/docs/en/main/future/vite#augmenting-load-context). I decided not to because I felt the code would provide a good explanation and help readers learn. Linking to external docs might be more confusing.

AdiRishi commented on content/pages/framework-guides/deploy-a-remix-site.md:
```diff
@@ -22,9 +22,9 @@ $ npm create cloudflare@latest my-remix-app -- --framework=remix
+Your Remix project will include a `functions/[[path]].ts` file. The `[[path]]` filename indicates that this file will handle requests to all incoming URLs. Refer to [Path segments](/pages/functions/routing/#dynamic-routes) to learn more.
 
-The `functions/[[path]].js` will not function as expected if you attempt to deploy your site before running `remix dev` or `remix build`.
+The `functions/[[path]].ts` will not function as expected if you attempt to deploy your site before running `remix vite:build`.
```
I seriously considered replacing `remix vite:build` here with `npm run build`. However I felt it was better to showcase the actual build command being run.

AdiRishi commented:
Okay, my `Binding Resources in Production` section may be outdated/unnecessary based on the latest `wrangler.toml` support released during dev week. I'll look into the changes as soon as I can and update the docs accordingly with new best practices

dario-piotrowicz commented on content/pages/_partials/_build-configuration.md:
```diff
@@ -38,7 +38,7 @@ build_configs:
     display_name: Remix
     build_command: npm run build
-    build_output_directory: public
+    build_output_directory: build
```
```suggestion
    build_output_directory: build/client
```

dario-piotrowicz replied:
(just `build` also deploys the `build/server` dir which is unnecessary)

AdiRishi replied:
That's so interesting, I remember testing `build`, seeing that it worked and not understanding why 😅 
Because locally you **have** to run `wrangler pages deploy ./build/client` for it to work at all.

dario-piotrowicz replied:
that's bizarre... maybe when you were using `build` by itself you were getting the correct server side rendered page but not all the client side chunks?

AdiRishi replied:
Ahh yes you're very correct. The SSR works but the client side chunks fail.

dario-piotrowicz commented on content/pages/framework-guides/deploy-a-remix-site.md:
````diff
@@ -62,79 +62,159 @@ For the complete guide to deploying your first site to Cloudflare Pages, refer t
+};
+```
+
+#### Correcting env types
````
If the user has created the application using C3 the following is a bit off since they can just run:
```sh
$ npm run build-cf-types
```

Could you clarify this?

AdiRishi replied:
ahh, I get why I got confused here. So there is some subtlety involved.

If anyone scaffolds this project using the default remix template (aka `npx create-remix@latest --template remix-run/remix/templates/cloudflare`) then they won't get a `worker-configuration.d.ts` file with their project. For them, their `load-context.ts` file will look like this

```ts
import { type PlatformProxy } from "wrangler";

// Empty interface, needs to be manually populated with the output of wrangler types
interface Env {}

type Cloudflare = Omit<PlatformProxy<Env>, "dispose">;

declare module "@remix-run/cloudflare" {
  interface AppLoadContext {
    cloudflare: Cloudflare;
  }
}
```

However, as you rightly pointed out, when you scaffold with `c3`, this file is setup to expect the `Env` interface that is defined in `worker-configuration.d.ts` which is auto-updated when you run `wrangler types`. As a point of note, there are two npm commands that do this, `npm run typegen` (provided by the remix template) and `npm run  build-cf-types` (provided by `c3`)

@dario-piotrowicz I'm leaning towards your recommendation, and simply telling users to run `npm run typegen` (or build-cf-types), rather than explaining the whole copy paste thing.
I also think I should probably make a PR back into Remix to update their template to also use `worker-configuration.d.ts`, since it seems like the more sensible option

dario-piotrowicz replied:
> I also think I should probably make a PR back into Remix to update their template to also use worker-configuration.d.ts, since it seems like the more sensible option

Yes, that's something that I've wanted to do for a bit but haven't really find the time to do so 😅

if you want please go right ahead, or I can take it on (and prioritize this 😅), whatever works best for you 🙂👍 

One thing to keep in mind, is that `wrangler types` accepts an `--env-interface` argument to update the interface name and can also write to a different output file, I would use these to make the interface and file less generic and more cloudflare-specific 🙂

AdiRishi replied:
> if you want please go right ahead, or I can take it on (and prioritize this 😅), whatever works best for you 🙂👍

Hey if you think you can finish it before the coming weekend go right ahead 😄 
I'll likely be busy with other work until Saturday

> One thing to keep in mind, is that wrangler types accepts an --env-interface argument to update the interface name and can also write to a different output file, I would use these to make the interface and file less generic and more cloudflare-specific 🙂

I'm not sure I follow. Is the idea to modify `load-context.ts` + the npm commands in a way that we write the generated types directly to that file rather than a seperate `.d.ts` file?

dario-piotrowicz commented on content/pages/framework-guides/deploy-a-remix-site.md:
```diff
@@ -62,79 +62,159 @@ For the complete guide to deploying your first site to Cloudflare Pages, refer t
-      {/* ... */}
-    </div>
-  );
+To bind resources in production you need to configure the bindings in the Cloudflare dashboard. Refer to the [Bindings](/pages/functions/bindings/) documentation to learn more.
```
this section seems overly simplified for me.... it doesn't include any information on how to use the bindings (via `context.cloudflare`)...

I can understand that you didn't want to repeat the same information provided in the local setup section...
But could you at least mention that the way to access/use the bindings is the same as mentioned above?

(for now, then when we restructure this guide we can improve this further)

dario-piotrowicz replied:
Thanks for addressing this 🙂

dario-piotrowicz commented:
> Okay, my `Binding Resources in Production` section may be outdated/unnecessary based on the latest `wrangler.toml` support released during dev week. I'll look into the changes as soon as I can and update the docs accordingly with new best practices

I wouldn't do that as the `wrangler.toml` support for Pages is in beta, so I would still wait before adding it to these docs

dario-piotrowicz commented on content/pages/framework-guides/deploy-a-remix-site.md:
````diff
@@ -62,79 +62,159 @@ For the complete guide to deploying your first site to Cloudflare Pages, refer t
+database_name = "<your-db-name>"
+database_id = "<your-db-id>"
+```
+3. Run `wrangler types` to generate the types for the bindings:
````
as per my comment above, assuming people scaffolded the app using C3 they could just run `npm run build-cf-types`

dario-piotrowicz reviewed (commented):
Thanks a lot @AdiRishi much appreciated 😄

I've left a few comments regarding small things that I think should be amended before we land the PR

But overall it looks great to me, thanks for such nice improvements 😄 ❤️

AdiRishi commented:
@dario-piotrowicz thank you for the review, you raised a lot of great points that I didn't consider 😅 
I think I've addressed all the issues raised. I opted to remove the `Correcting env types` section in favour of an alert type box. Seemed more appropriate given the size and context.

Thanks again for being so quick and working with me to get this merged. I hope to make a lot more contributions in the future 🧡

dario-piotrowicz commented on content/pages/framework-guides/deploy-a-remix-site.md:
```diff
@@ -62,79 +62,106 @@ For the complete guide to deploying your first site to Cloudflare Pages, refer t
+
+To bind resources in production you need to configure the bindings in the Cloudflare dashboard. Refer to the [Bindings](/pages/functions/bindings/) documentation to learn more.
+
+Once you have configured the bindings in the Cloudflare dashboard, the proxies are then available within `context.cloudflare.env` in your `loader` or `action` functions as shown [above](/pages/framework-guides/deploy-a-remix-site/#binding-resources-in-local-development).
```
```suggestion
Once you have configured the bindings in the Cloudflare dashboard, the proxies are then available within `context.cloudflare.env` in your `loader` or `action` functions as shown [above](#binding-resources-in-local-development).
```

dario-piotrowicz commented on content/pages/framework-guides/deploy-a-remix-site.md:
````diff
@@ -62,79 +62,106 @@ For the complete guide to deploying your first site to Cloudflare Pages, refer t
+};
+```
+
+{{<Aside type="warning" header="Correcting the env type">}}
````
I'm not sure if this needs to be a warning 🤔

```suggestion
{{<Aside header="Correcting the env type">}}
```

dario-piotrowicz reviewed (approved):
Looks good to me, thanks again @AdiRishi ❤️

dario-piotrowicz commented:
> Thanks again for being so quick and working with me to get this merged.

My pleasure! 😄

>  I hope to make a lot more contributions in the future 🧡

Great! They're always welcome! 🧡

dario-piotrowicz commented:
@AdiRishi just wanted to let you know that technical writer that handles these pages is out of office for the next few days, so merging this PR can unfortunately get delayed for a bit 😓

AdiRishi commented:
> @AdiRishi just wanted to let you know that technical writer that handles these pages is out of office for the next few days, so merging this PR can unfortunately get delayed for a bit 😓

All good! I just figured we'd want to merge sooner rather than later due to increased activity after dev week. Don't want newcomers to get too confused haha

deadlypants1973 commented on content/pages/framework-guides/deploy-a-remix-site.md:
```diff
@@ -62,79 +62,106 @@ For the complete guide to deploying your first site to Cloudflare Pages, refer t
+
+## Example: Accessing your D1 database in a Remix application
+
+Let's put the steps above into practice by binding and querying a D1 database in a Remix application.
```
```suggestion
As an example, you will bind and query a D1 database in a Remix application.
```

deadlypants1973 commented:
closes #13637

dario-piotrowicz commented:
@AdiRishi merged! 😄 thanks again for the nice improvements! 💪
